presto-main/src/main/java/com/facebook/presto/operator/aggregation/AggregationUtils.java

```diff
@@ -126,22 +125,24 @@ public static void mergeRegressionState(RegressionState state, RegressionState o
         state.setSumXSquare(state.getSumXSquare() + otherState.getSumXSquare());
     }
 
-    public static Type getOutputType(@Nullable Method outputFunction, AccumulatorStateSerializer<?> serializer, TypeManager typeManager)
+    public static TypeSignature getOutputTypeSignature(@Nullable Method outputFunction)
     {
-        if (outputFunction == null) {
-            return serializer.getSerializedType();
-        }
-        else {
-            return typeManager.getType(TypeSignature.parseTypeSignature(outputFunction.getAnnotation(OutputFunction.class).value()));
-        }
+        checkArgument(outputFunction != null);
+        return TypeSignature.parseTypeSignature(outputFunction.getAnnotation(OutputFunction.class).value());
     }
 
+    @Deprecated
     public static String generateAggregationName(String baseName, Type outputType, List<Type> inputTypes)
     {
+        return generateAggregationName(baseName, outputType.getTypeSignature(), inputTypes.stream().map(x -> x.getTypeSignature()).collect(Collectors.toList()));
+    }
+
+    public static String generateAggregationName(String baseName, TypeSignature outputType, List<TypeSignature> inputTypes)
```

Comment:
Instead of deprecating the other form of this, can you move this change to a separate commit and just remove the other function?